app/adapters/weather/client.rb

```diff
@@ -0,0 +1,33 @@
+module Weather
+  class Client
+    include HTTParty
+    base_uri "https://dataservice.accuweather.com"
+
+    def initialize
+      @apikey = ENV.fetch("API_Key")
+      @town_id = ENV.fetch("Town_ID")
+    end
+
+    def fetch_daily_forecast(query = {})
+      self.class.get("/forecasts/v1/daily/1day/#{@town_id}", query: query_prepare(query))
+    end
+
+    def fetch_current_conditions_historical
+      self.class.get("/currentconditions/v1/#{@town_id}/historical/24", query: hashed_apikey)
+    end
+
+    def fetch_current_conditions
+      self.class.get("/currentconditions/v1/#{@town_id}", query: hashed_apikey)
+    end
+
+    private
+
+    def hashed_apikey
+      { apikey: @apikey }
+    end
```

Comment:
Возможно в данном конкретном случае проблем и не будет, но я вижу что этот метод в коде используется более одного раза, а значит на всякий случай лучше его захешировать

```suggestion
    def hashed_apikey
      @hashed_apikey ||= { apikey: @apikey }
    end
```